lib/msf/core/auxiliary/ubiquiti.rb

```diff
@@ -243,7 +243,7 @@ def process_settings(lines, credential_data)
           cred[:private_data] = admin_password_hash
           cred[:private_type] = :nonreplayable_hash
           login = create_credential_and_login(cred)
```

Comment:
It seems like the tests fail because the tests originally mocked out the `create_credential_and_login` call, which returns nil:

https://github.com/rapid7/metasploit-framework/blob/8772f3dc7749964456b03c6e8bf772771e3c006b/spec/lib/msf/core/auxiliary/ubiquiti_spec.rb#L364-L378

And in a later commit https://github.com/rapid7/metasploit-framework/pull/12594/commits/f58054b24d9c6e0144f06ae5a11743607ef0af22 - the implementation was modified to make use of this return value

The initial thought was to update the test, but on further inspection it seems like `create_credential_and_login` can return nil in the scenario of not having an active database. So this seems like the right solution 🤞

Reply:
That test wasn't necessarily supposed to mock that, but I didn't know how to make it work correctly.  https://github.com/rapid7/metasploit-framework/pull/12594#issuecomment-575200061
looks like there was possibly a deleted comment in there (since my comment doesn't seem to have context).

However, this is a solution that works, so i'm all for it.  will land it shortly